scantext/views.py

```diff
@@ -40,25 +40,64 @@
 def license_scanview(request):
     form = LicenseScanForm()
     if request.method == "POST":
-        form = LicenseScanForm(request.POST)
+        form = LicenseScanForm(request.POST, request.FILES)
         if form.is_valid():
-            text = form.cleaned_data["input_text"]
+            input_text = form.cleaned_data["input_text"]
+            input_file = request.FILES.get("input_file", False)
+            if not len(input_text) and not input_file:
+                message = "Please provide some text or a text file to scan."
+                return render(
+                    request,
+                    "scantext/license_scan_form.html",
+                    {
+                        "form": LicenseScanForm(),
+                        "input_error": message,
+                    },
+                )
+
             # The flush in tempfile is required to ensure that the content is
             # written to the disk before it's read by get_licenses function
-            with tempfile.NamedTemporaryFile(
-                mode="w",
-            ) as temp_file:
-                temp_file.write(text)
-                temp_file.flush()
-                expressions = get_licenses(
-                    location=temp_file.name,
+            if len(input_text):
+                with tempfile.NamedTemporaryFile(
+                    mode="w",
+                ) as temp_file:
+                    temp_file.write(input_text)
+                    temp_file.flush()
+                    expressions = get_licenses(
+                        location=temp_file.name,
+                    )
```

Comment:
Breaking short line does not help readability. This would be better as a one-liner.

Reply:
sure, will make it one-liner